hydromt_wflow/wflow.py

```diff
@@ -4567,13 +4567,21 @@ def write_grid(
         if not isdir(dirname(fn)):
             os.makedirs(dirname(fn))
         self.logger.info(f"Write grid to {fn}")
-        mask = ds_out[self._MAPS["basins"]] > 0
-        for v in ds_out.data_vars:
-            # nodata is required for all but boolean fields
-            if ds_out[v].dtype != "bool":
-                ds_out[v] = ds_out[v].where(mask, ds_out[v].raster.nodata)
+        ds_out = self._mask_grid_to_basins(ds_out)
+
         ds_out.to_netcdf(fn, encoding=encoding)
 
+    def _mask_grid_to_basins(self, ds_out):
```

Comment:
Nice! maybe you can `basin_layer_name` as an argument and move this function to utils rather than model?